src/psyclone/psyGen.py

```diff
@@ -2909,11 +2927,30 @@ def coloured_text(self):
         '''
         return colored("KernCall", SCHEDULE_COLOUR_MAP["KernCall"])
 
-    def gen_code(self, parent):
+    def gen_code(self, parent, arguments=None):
+        '''
+        Generates the f2pygen AST of the Fortran for this kernel call and
+        writes the kernel itself to file if it has been transformed.
+
+        :param parent: The parent of this kernel call in the f2pygen AST.
+        :type parent: :py:calls:`psyclone.f2pygen.LoopGen`
+        :param arguments: The list of arguments to pass to the kernel.
+        :type arguments: list of str.
+        '''
         from psyclone.f2pygen import CallGen, UseGen
-        parent.add(CallGen(parent, self._name, self._arguments.arglist))
-        parent.add(UseGen(parent, name=self._module_name, only=True,
-                          funcnames=[self._name]))
+
+        # If kernel has been transformed then we rename it and write it to file
+        self.update()
```

Comment:
Should we avoid writing to file if we are (module) inlining?

Reply:
Oh yes, good point. I was initially worried that this would mean that we wouldn't be able to check whether or not a subsequent transformation of another instance of the same kernel gave the same result (which we forbid if the kernel-renaming scheme is "single"). However, I then realised that we only worry about that because we need to ensure that the correct versions of the kernels are called. If a kernel is module in-lined then it's scope is local and we don't have to worry about it clashing with other instances.